sdk/python/kfp/components/structures/components.json_schema.json

```diff
@@ -0,0 +1,402 @@
+        {"$ref": "#/definitions/ConcatPlaceholder"},
+        {"$ref": "#/definitions/IfPlaceholder"}
+      ]
+    },
+
+    "CommandlineArgumentOrArrayType": {
+      "oneOf": [
+        {"$ref": "#/definitions/CommandlineArgumentType"},
+        {"type": "array", "items": {"$ref": "#/definitions/CommandlineArgumentType"}}
+      ]
+    },
+
+    "ConcatPlaceholder": {
+      "description": "Represents the command-line argument placeholder that will be replaced at run-time by the concatenated values of its items.",
+      "type": "object",
+      "required": ["concat"],
+      "properties": {
+        "concat" : {
+          "description": "Items to concatenate",
+          "type": "array",
+          "items": {"$ref": "#/definitions/CommandlineArgumentType"}
+        }
+      },
+      "additionalProperties": false
+    },
+
+    "IsPresentPlaceholder": {
+      "description": "Represents the command-line argument placeholder that will be replaced at run-time by a boolean value specifying whether the caller has passed an argument for the specified optional input.",
+      "type": "object",
+      "properties": {
+        "isPresent": {
+          "description": "Name of the input.",
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+
+    "IfConditionArgumentType": {
+      "oneOf": [
+        {"$ref": "#/definitions/IsPresentPlaceholder"},
+        {"type": "boolean"},
+        {"type": "string"},
+        {"$ref": "#/definitions/InputValuePlaceholder"}
+      ]
+    },
+
+    "IfPlaceholder": {
+      "description": "Represents the command-line argument placeholder that will be replaced at run-time by a boolean value specifying whether the caller has passed an argument for the specified optional input.",
+      "type": "object",
+      "required": ["if"],
+      "properties": {
+        "if" : {
+          "type": "object",
+          "required": ["cond", "then"],
+          "properties": {
+            "cond": {"$ref": "#/definitions/IfConditionArgumentType"},
+            "then": {"$ref": "#/definitions/CommandlineArgumentOrArrayType"},
+            "else": {"$ref": "#/definitions/CommandlineArgumentOrArrayType"}
+          }
+        }
+      }
+    },
+
+    "ContainerSpec": {
+      "type": "object",
+      "required": ["image"],
+      "properties": {
+        "image": {
+          "description": "Docker image name.",
+          "type": "string"
+        },
+        "command": {
+          "description": "Entrypoint array. Not executed within a shell. The docker image's ENTRYPOINT is used if this is not provided.",
+          "type": "array",
+          "items": {"$ref": "#/definitions/CommandlineArgumentType"}
+        },
+        "args": {
+          "description": "Arguments to the entrypoint. The docker image's CMD is used if this is not provided.",
+          "type": "array",
+          "items": {"$ref": "#/definitions/CommandlineArgumentType"}
+        },
+        "env": {
```

Comment:
For GCP component, I want to pass additional env from pod metadata. Why not just use k8s env definition here?

Reply:
+1

Reply:
Can you please give an example? It's always helpful to me to see the concrete example to have a sense of what different people need.

Do you just want EnvFrom? Is the configuration static (always the same for all component instances (tasks))?

Reply:
JFYI, I was torn between two requests - to keep the spec minimal and to include everything the Kubernetes supports. I think that I have both - simple and portable `ContainerSpec` plus full k8s `TaskSpec.k8sContainerOptions` + `TaskSpec.k8sPodOptions`. What do you think?

Speaking about env, you can set it in `TaskSpec.k8sContainerOptions.env`.

Reply:
Talked with Hongye offline.

For this particular case we can use Argo's `{{pod.name}}` placeholder of an analog of it.

```
container:
  env:
    POD_NAME: '{{pod.name}}'
```
or
```
container:
  env:
    POD_NAME: {podMetadata: 'name'}
```